src/main/java/it/gov/pagopa/debtposition/scheduler/HistoricizationScheduler.java

```diff
@@ -0,0 +1,192 @@
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnit;
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableClientBuilder;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.TableErrorCode;
+import com.azure.data.tables.models.TableServiceException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import it.gov.pagopa.debtposition.entity.PaymentOption;
+import it.gov.pagopa.debtposition.entity.PaymentPosition;
+import it.gov.pagopa.debtposition.repository.PaymentPositionRepository;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
+
+@Component
+@Slf4j
+@ConditionalOnProperty(name = "cron.job.schedule.historicization.enabled", matchIfMissing = true)
+@NoArgsConstructor
+public class HistoricizationScheduler {
+
+	private static final String LOG_BASE_HEADER_INFO = "[OperationType: %s] - [ClassMethod: %s] - [MethodParamsToLog: %s]";
+    private static final String CRON_JOB = "CRON JOB";
+    private static final String METHOD = "manageDebtPositionsToHistoricize";
+    @Getter
+    private Thread threadOfExecution;
+    
+    
+    
+    // extraction params
+    @Value("${cron.job.schedule.extraction.history.query:SELECT pp FROM PaymentPosition pp WHERE pp.status IN ('PAID', 'REPORTED', 'INVALID', 'EXPIRED') AND pp.lastUpdatedDate < ?1}")
+    private String extractionQuery;
+    @Value("${cron.job.schedule.extraction.history.query.interval:365}")
+    private short extractionInterval;
+    // extraction params: pagination mode
+    @Value("${cron.job.schedule.extraction.history.query.paginated:true}")
+    private boolean paginationMode;
+    @Value("${cron.job.schedule.extraction.history.query.count:SELECT count(pp.id) FROM PaymentPosition pp WHERE pp.status IN ('PAID', 'REPORTED', 'INVALID', 'EXPIRED') AND pp.lastUpdatedDate < ?1}")
+    private String countExtractionQuery;
+    @Value("${cron.job.schedule.extraction.history.query.page.size:100000}")
+    private int pageSize;
+    
+    // azure storage params
+    @Value("${azure.archive.storage.connection}")
+    private String archiveStorageConnection;
+    @Value("${azure.archive.storage.table.po:pagopadweugpsarchivesapaymentoptiontable}")
```

Comment:
oki, shall not contains env `pagopadweugpsarchives`

Reply:
done